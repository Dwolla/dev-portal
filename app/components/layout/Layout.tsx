/* eslint-disable jsx-a11y/click-events-have-key-events */
/* eslint-disable jsx-a11y/no-noninteractive-element-interactions */
/* eslint-disable jsx-a11y/no-static-element-interactions */
import Head from "next/head";
import { css, Global } from "@emotion/core";
import styled from "@emotion/styled";
import classnames from "classnames";
import { useState, useEffect } from "react";
import Link from "next/link";
import {
  disableBodyScroll,
  enableBodyScroll,
  clearAllBodyScrollLocks,
} from "body-scroll-lock";
import { useRouter } from "next/router";
import SideNav, { SideNavLinkProps } from "./SideNav"; // eslint-disable-line no-unused-vars
import { POPPINS, ROBOTO } from "../typography";
import { BOX_SHADOW_6 } from "../shadowDepths";
import {
  GREY_2,
  GREY_8,
  WHITE_PRIMARY,
  ORANGE_PRIMARY,
  PURPLE_PRIMARY_BUTTON,
  PURPLE_PRIMARY_HOVER,
  PURPLE_PRIMARY_ACTIVE,
  HEADLINE_TEXT,
} from "../colors";
import { breakUp, breakDown } from "../breakpoints";
import TopBar, { TopBarProps, TOP_BAR_HEIGHT } from "./TopBar"; // eslint-disable-line no-unused-vars
import Footer, { FooterLink } from "./Footer"; // eslint-disable-line no-unused-vars
import APIStatusBar from "./APIStatusBar";
import { ReactComponent as CloseIcon } from "../../../assets/images/component-icons/close.svg";
import dwollaDevLogo from "../../../assets/images/dwolla-developers-logo.svg";

export const LEFT_SIDEBAR_PADDING_X = "20px";

const LeftSidebar = styled.div`
  position: fixed;
  top: 0;
  right: 0;
  bottom: 0;
  left: 0;
  overflow-y: auto;
  display: flex;
  flex-direction: column;
  justify-content: space-between;
  border-right: 1px solid ${GREY_2};
  transition: opacity 300ms;
  opacity: 0;
  pointer-events: none;
  z-index: 9999;

  @media (${breakUp("lg")}) {
    right: 75%;
    opacity: 1;
    pointer-events: auto;
  }

  &.toggled {
    opacity: 1;
    pointer-events: auto;
  }

  /* Hides sidebar on smaller screens when not toggled. Not tabbable when not in view  */
  &.visuallyHidden {
    @media (${breakDown("md")}) {
      visibility: hidden;
    }
  }
`;

const LogoWrapper = styled.div`
  padding: 12px ${LEFT_SIDEBAR_PADDING_X};
  background: white;
  flex-shrink: 1;
  display: flex;
  align-items: center;
  justify-content: space-between;

  @media (${breakDown("md")}) {
    border-bottom: 1px solid ${GREY_2};
    height: ${TOP_BAR_HEIGHT}px;
    min-height: ${TOP_BAR_HEIGHT}px;
  }
`;

const SideNavWrapper = styled.div`
  flex-grow: 1;
  overflow: hidden;
  background: white;
  display: flex;
  flex-direction: column;
`;

const StyledCloseIcon = styled.div`
  cursor: pointer;
  svg {
    * {
      fill: ${ORANGE_PRIMARY};
    }
  }

  :hover,
  &.isFocused {
    svg {
      * {
        fill: ${GREY_8};
      }
    }
  }

  :focus {
    outline: none;
  }

  @media (${breakUp("lg")}) {
    display: none;
  }
`;

const MainArea = styled.div`
  @media (${breakUp("lg")}) {
    margin-left: 25%;
  }
`;

const ContentArea = styled.div`
  @media (${breakUp("lg")}) {
    margin-right: 40px;
  }

  /* Hides ContentArea on smaller screens when SideBar is toggled. Not tabbable when not in view  */
  &.visuallyhidden {
    @media (${breakDown("md")}) {
      visibility: hidden;
    }
  }
`;

const TopBarWrapper = styled.div`
  position: sticky;
  top: 0;
  z-index: 99;
`;

const FooterWrapper = styled.div`
  border-top: 1px solid ${GREY_2};
  background: ${WHITE_PRIMARY};
  z-index: 9;

  @media (${breakUp("xxl")}) {
    padding-right: 40px;
  }

  /* Hides FooterArea on smaller screens when SideBar is toggled. Not tabbable when not in view  */
  &.visuallyhidden {
    @media (${breakDown("md")}) {
      visibility: hidden;
    }
  }
`;

export default function Layout({
  children,
  pages,
  sideNavLinks,
  footerLinks,
  footerLegal,
  topBarProps,
  apiStatus,
}: {
  children: JSX.Element;
  pages: Page[];
  sideNavLinks: SideNavLinkProps[];
  footerLinks: Record<string, FooterLink[]>;
  footerLegal: {
    title: string;
    operatorDescription: string;
    veridianDescription: string;
    metaBankDescription: JSX.Element;
    rtpDescription: string;
  };
  topBarProps: TopBarProps;
  apiStatus: APIStatus;
}) {
  const [sidebarToggled, setSidebarToggled] = useState(false);
  const [isFocused, setIsFocused] = useState(false);

  useEffect(() => {
    if (document) {
      enableBodyScroll(document.querySelector("#body-scroll-lock-side-nav"));
      if (sidebarToggled) {
        disableBodyScroll(document.querySelector("#body-scroll-lock-side-nav"));
      }
    }
    return () => {
      clearAllBodyScrollLocks();
    };
  }, [sidebarToggled]);

  const showSidebar = () => setSidebarToggled(true);
  const hideSidebar = () => setSidebarToggled(false);

  function onKeydown(e) {
    if (e.key === "Enter") hideSidebar();
  }

  const router = useRouter();

  return (
    <>
      <Global
        styles={css`
          * {
            box-sizing: border-box;
          }

          #dwollaCookieConsent {
            font-family: ${ROBOTO};
            color: ${HEADLINE_TEXT};
            line-height: 22px;

            a {
              color: ${ORANGE_PRIMARY};
              text-decoration: none;

              &:hover {
                text-decoration: underline;
              }
            }

            div {
              button {
                font-family: ${POPPINS};
                background-color: ${PURPLE_PRIMARY_BUTTON};
                border-color: transparent;

                :hover,
                :focus {
                  background-color: ${PURPLE_PRIMARY_HOVER};
                  box-shadow: ${BOX_SHADOW_6};
                }

                :active {
                  background-color: ${PURPLE_PRIMARY_ACTIVE};
                }
              }
            }
          }
        `}
      />

      <Head>
        <title>Dwolla Developers</title>
        <meta name="viewport" content="initial-scale=1.0, width=device-width" />
        <link
          rel="stylesheet"
          href="https://cdnjs.cloudflare.com/ajax/libs/modern-normalize/0.5.0/modern-normalize.min.css"
        ></link>
        <link
          href="https://fonts.googleapis.com/css?family=Poppins:100,100i,200,200i,300,300i,400,400i,500,500i,600,600i,700,700i,800,800i,900,900i|Roboto:100,100i,300,300i,400,400i,500,500i,700,700i,900,900i&display=swap"
          rel="stylesheet"
        ></link>
        <link rel="canonical" href={router?.pathname} />
        <script
          type="text/javascript"
          src="//js.hsforms.net/forms/v2.js"
        ></script>
      </Head>

      <LeftSidebar
        className={classnames(sidebarToggled ? "toggled" : "visuallyHidden")}
      >
        <LogoWrapper>
          <Link href="/">
            <a>
              <img
                src={dwollaDevLogo}
                alt="Dwolla Developers Logo"
                css={css`
                  @media (${breakDown("md")}) {
                    max-height: 100%;
                  }

                  @media (${breakDown("xs")}) {
                    max-width: 115px;
                  }
                `}
              />
            </a>
          </Link>

          <StyledCloseIcon
            tabIndex={0}
            className={classnames({ isFocused })}
            onClick={hideSidebar}
            onKeyDown={(keyPress) => onKeydown(keyPress)}
            onFocus={() => setIsFocused(true)}
            onBlur={() => setIsFocused(false)}
          >
            <CloseIcon width={14} />
          </StyledCloseIcon>
        </LogoWrapper>

        <SideNavWrapper>
          <SideNav
            sectionLinks={sideNavLinks}
            pages={pages}
            mobileItems={topBarProps}
          />
        </SideNavWrapper>

        <APIStatusBar apiStatus={apiStatus} />
      </LeftSidebar>

      <MainArea>
        <TopBarWrapper>
          <TopBar {...topBarProps} onHamburgerClick={showSidebar} />
        </TopBarWrapper>

        <ContentArea className={classnames({ visuallyhidden: sidebarToggled })}>
          {children}
        </ContentArea>

        <FooterWrapper
          className={classnames({ visuallyhidden: sidebarToggled })}
        >
          <Footer links={footerLinks} legal={footerLegal} />
        </FooterWrapper>
      </MainArea>
    </>
  );
}
