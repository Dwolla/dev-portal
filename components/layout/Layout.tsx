import Head from "next/head";
import { css, Global } from "@emotion/core";
import styled from "@emotion/styled";
import classnames from "classnames";
import SideNav, { SideNavLinkProps } from "./SideNav"; // eslint-disable-line no-unused-vars
import { GREY_2 } from "../colors";
import ApiStatus from "./ApiStatus";
import TopBar from "./TopBar";
import Footer, { FooterLink } from "./Footer"; // eslint-disable-line no-unused-vars
import { Page } from "../../modules/pages"; // eslint-disable-line no-unused-vars

// proptypes

interface LayoutProps {
  children: JSX.Element;
  pages: Page[];
  sideNavLinks: SideNavLinkProps[];
  footerLinks: Record<string, FooterLink[]>;
  topBarChildren: JSX.Element | JSX.Element[];
}

// components

export const LEFT_SIDEBAR_PADDING_X = "20px";

const LeftSidebar = styled.div`
  position: fixed;
  top: 0;
  right: 10%;
  bottom: 0;
  left: 0;
  overflow-y: scroll;
  display: flex;
  flex-direction: column;
  justify-content: space-between;
  border-right: 1px solid ${GREY_2};
  transform: translate3d(-100%, 0, 0);
  transition: transform 0.2s ease-in-out;

  @media (min-width: 980px) {
    right: 75%;
    transform: translate3d(0, 0, 0);
  }

  &.toggled {
    transform: translate3d(0, 0, 0);

    @media (min-width: 980px) {
      right: 75%;
    }
  }
`;

const LogoWrapper = styled.div`
  padding: 12px ${LEFT_SIDEBAR_PADDING_X};
  background: white;
  flex-shrink: 1;
`;

const SideNavWrapper = styled.div`
  flex-grow: 1;
  overflow: hidden;
  background: white;
  padding-bottom: 60px;
`;

const MainArea = styled.div`
  @media (min-width: 980px) {
    margin-left: 25%;
  }
`;

export default function Layout(props: LayoutProps) {
  const sidebarToggled = false;

  return (
    <>
      <Global
        styles={css`
          * {
            box-sizing: border-box;
          }
        `}
      />

      <Head>
        <title>Dwolla Developers</title>
        <meta name="viewport" content="initial-scale=1.0, width=device-width" />
        <link
          rel="stylesheet"
          href="https://cdnjs.cloudflare.com/ajax/libs/modern-normalize/0.5.0/modern-normalize.min.css"
        ></link>
        <link
          href="https://fonts.googleapis.com/css?family=Poppins:100,100i,200,200i,300,300i,400,400i,500,500i,600,600i,700,700i,800,800i,900,900i|Roboto:100,100i,300,300i,400,400i,500,500i,700,700i,900,900i&display=swap"
          rel="stylesheet"
        ></link>
      </Head>

      <LeftSidebar className={classnames({ toggled: sidebarToggled })}>
        <LogoWrapper>
          <img
            src="/images/dwolla-developers-logo.png"
            alt="Dwolla Developers Logo"
          />
        </LogoWrapper>

        <SideNavWrapper>
          <SideNav links={props.sideNavLinks} pages={props.pages} />
        </SideNavWrapper>

        <ApiStatus />
      </LeftSidebar>

      <MainArea>
        <TopBar>{props.topBarChildren}</TopBar>

        {props.children}

        <Footer links={props.footerLinks} />
      </MainArea>
    </>
  );
}
