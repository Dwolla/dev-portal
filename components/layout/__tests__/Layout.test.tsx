import renderer from "react-test-renderer";
import Layout from "../Layout";
import Pages from "../../../modules/pages";

const mockPath = "/";
jest.mock("next/router", () => ({
  useRouter() {
    return {
      route: mockPath,
      pathname: mockPath,
      query: "",
      asPath: mockPath,
    };
  },
}));

const SIDE_NAV_LINKS = [
  {
    href: "/",
    iconSrc: "/images/home-icon.svg",
    isSection: false,
    text: "Home",
  },
  {
    href: "/docs/guides",
    iconSrc: "/images/guides-icon.svg",
    isSection: true,
    text: "Guides",
  },
  {
    href: "/docs/concepts",
    iconSrc: "/images/concepts-icon.svg",
    isSection: true,
    text: "Concepts",
  },
  {
    href: "/sdks-tools",
    iconSrc: "/images/sdks-tools-icon.svg",
    isSection: false,
    text: "SDKs & Tools",
  },
];

const FOOTER_LINKS = {
  Discover: [
    { text: "Send", href: "#discover-send" },
    { text: "Receive", href: "#discover-receive" },
    { text: "Send & Receive", href: "#discover-send-receive" },
    { text: "Facilitate", href: "#facilitate" },
    { text: "Me to Me", href: "#me-to-me" },
  ],
  Concepts: [
    { text: "Basics", href: "#concepts-basics" },
    { text: "Essentials", href: "#concepts-essentials" },
    { text: "Integrations", href: "#concepts-integrations" },
  ],
  Guides: [
    { text: "Prerequisites", href: "#guides-prerequisites" },
    { text: "Get Building", href: "#guides-get-building" },
    { text: "Features", href: "#features" },
  ],
  "API Reference": [
    { text: "Documentation", href: "#api-reference-documentation" },
    { text: "SDK Support", href: "api-reference-sdk-support" },
    { text: "What does the Dwolla API do?", href: "#api-reference-what-it-do" },
  ],
  "Tools and SDKs": [
    { text: "Ruby", href: "#tools-and-sdks-ruby" },
    { text: "Python", href: "#tools-and-sdks-python" },
    { text: "PHP", href: "#tools-and-sdks-php" },
    { text: "C#", href: "#tools-and-sdks-csharp" },
    { text: "Java", href: "#tools-and-sdks-java" },
    { text: "JavaScript", href: "#tools-and-sdks-javascript" },
  ],
};

const TOP_BAR_PROPS = {
  button: { text: "Get API Keys" },
  links: [
    { text: "API Docs", href: "https://docs.dwolla.com", external: true },
    { text: "Changelog", href: "/changelog" },
  ],
};

test("Layout", () => {
  const tree = renderer
    .create(
      <Layout
        sideNavLinks={SIDE_NAV_LINKS}
        footerLinks={FOOTER_LINKS}
        topBarProps={TOP_BAR_PROPS}
        pages={Pages.all()}
        apiStatus={{ indicator: "none", description: "apiStatus description" }}
      >
        <div>page</div>
      </Layout>
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});
