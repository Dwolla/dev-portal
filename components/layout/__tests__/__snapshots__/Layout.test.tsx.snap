// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Layout 1`] = `
Array [
  <div
    className=" css-78it7j-LeftSidebar eu5n9bq0"
  >
    <div
      className="css-3xwi1s-LogoWrapper eu5n9bq1"
    >
      <img
        alt="Dwolla Developers Logo"
        src="/images/dwolla-developers-logo.png"
      />
    </div>
    <div
      className="css-11xrmrt-SideNavWrapper eu5n9bq2"
    >
      <div
        className="css-1gqrwdk-Container e10mac9s0"
      >
        <div
          className=" css-1f0sx94-Slide e10mac9s1"
        >
          <div
            className="css-p3a8sa-SlidePane e10mac9s2"
          >
            <div
              className="css-1klu905-SectionLinksWrapper e10mac9s8"
            >
              <a
                className="css-3bamg-SectionLink-SectionLink"
                href="/"
                onClick={[Function]}
                onMouseEnter={[Function]}
              >
                <div>
                  <img
                    alt="Home"
                    src="/images/home-icon.svg"
                  />
                </div>
                Home
              </a>
              <a
                className="css-3bamg-SectionLink-SectionLink"
                href="/docs/guides"
                onClick={[Function]}
                onMouseEnter={[Function]}
              >
                <div>
                  <img
                    alt="Guides"
                    src="/images/guides-icon.svg"
                  />
                </div>
                Guides
              </a>
              <a
                className="css-3bamg-SectionLink-SectionLink"
                href="/docs/concepts"
                onClick={[Function]}
                onMouseEnter={[Function]}
              >
                <div>
                  <img
                    alt="Concepts"
                    src="/images/concepts-icon.svg"
                  />
                </div>
                Concepts
              </a>
              <a
                className="css-3bamg-SectionLink-SectionLink"
                href="/sdks-tools"
                onClick={[Function]}
                onMouseEnter={[Function]}
              >
                <div>
                  <img
                    alt="SDKs & Tools"
                    src="/images/sdks-tools-icon.svg"
                  />
                </div>
                SDKs & Tools
              </a>
            </div>
          </div>
          <div
            className="css-p3a8sa-SlidePane e10mac9s2"
          >
            <div
              className="css-mhn8j-StickyTop e10mac9s3"
            >
              <div
                className="css-1klu905-SectionLinksWrapper e10mac9s8"
              >
                <a
                  className="css-3bamg-SectionLink-SectionLink"
                  href="/"
                  onClick={[Function]}
                  onMouseEnter={[Function]}
                >
                  <div>
                    <img
                      alt="Back Home"
                      src="/images/home-icon.svg"
                    />
                  </div>
                  Back Home
                </a>
              </div>
            </div>
            <ul
              className="css-1ekxq2g-CategoriesNav e10mac9s4"
            />
          </div>
        </div>
      </div>
    </div>
    <a
      className="css-8aopii-ContainerLink e1nkb8hi0"
      href="https://status.dwolla.com"
      target="_blank"
    >
      <span
        className="indicator-none css-1tmj2bz-StatusIcon e1nkb8hi1"
      />
      apiStatus description
    </a>
  </div>,
  <div
    className="css-173gxmw-MainArea eu5n9bq3"
  >
    <div
      className="css-18nswzc-TopBarWrapper eu5n9bq5"
    >
      <div
        className="css-18kqr8q-Container e1xhgpyy0"
      >
        <a
          className=" css-1gmbmq9-StyledLink e1xhgpyy3"
          href="https://docs.dwolla.com"
          onClick={[Function]}
          onMouseEnter={[Function]}
          target="_blank"
        >
          <span
            className="text"
          >
            API Docs
          </span>
          <img
            alt="Open in new tab"
            className="icon"
            height={13}
            src="test-image-stub"
            width={13}
          />
        </a>
        <a
          className=" css-1gmbmq9-StyledLink e1xhgpyy3"
          href="/changelog"
          onClick={[Function]}
          onMouseEnter={[Function]}
        >
          <span
            className="text"
          >
            Changelog
          </span>
        </a>
        <div
          className="css-zhv85h-SelectWrapper e1xhgpyy2"
        >
          <div
            className=" css-k377xg-container"
            onKeyDown={[Function]}
          >
            <div
              className=" css-1xxipdn-control"
              onMouseDown={[Function]}
              onTouchEnd={[Function]}
            >
              <div
                className=" css-86rg1c-ValueContainer"
              >
                <div
                  className=" css-1me8bqy-singleValue"
                >
                  
                </div>
                <input
                  aria-autocomplete="list"
                  className="css-62g3xt-dummyInput"
                  disabled={false}
                  id="react-select-2-input"
                  onBlur={[Function]}
                  onChange={[Function]}
                  onFocus={[Function]}
                  readOnly={true}
                  tabIndex="0"
                  value=""
                />
              </div>
              <div
                className=" css-1hb7zxy-IndicatorsContainer"
              >
                <span
                  className=" css-43ykx9-indicatorSeparator"
                />
                <div
                  aria-hidden="true"
                  className=" css-unjupu-indicatorContainer"
                  onMouseDown={[Function]}
                  onTouchEnd={[Function]}
                >
                  <img
                    alt=""
                    className=" css-vfym8x-StyledDefaultDropdownIcon e1oqaitj0"
                    src="test-image-stub"
                  />
                </div>
              </div>
            </div>
          </div>
        </div>
        <div
          className="css-1px9jcr-ButtonWrapper e1xhgpyy1"
        >
          <button
            className="primary standard css-1eto6r1-ButtonStyle e12ysvgk0"
            type="button"
          >
            Get API Keys
          </button>
        </div>
      </div>
    </div>
    <div
      className="css-1uiuud6-ContentArea eu5n9bq4"
    >
      <div
        className="css-hr4r9m-ContentWrapper eu5n9bq7"
      >
        <div>
          page
        </div>
      </div>
      <div
        className="css-f0jn62-OnThisPageWrapper eu5n9bq6"
      />
    </div>
    <div
      className="css-cdxnl8-FooterWrapper eu5n9bq8"
    >
      <div
        className="css-e2rgnw-Container en8mxva0"
      >
        <div
          className="css-872axt-Group en8mxva1"
        >
          <h3
            className="css-p408ob-Heading en8mxva2"
          >
            Discover
          </h3>
          <ul
            className="css-499foe-List en8mxva3"
          >
            <li
              className="css-52hjeb-Item en8mxva4"
            >
              <a
                className="css-irwgy-ListLink en8mxva5"
                href="#discover-send"
              >
                Send
              </a>
            </li>
            <li
              className="css-52hjeb-Item en8mxva4"
            >
              <a
                className="css-irwgy-ListLink en8mxva5"
                href="#discover-receive"
              >
                Receive
              </a>
            </li>
            <li
              className="css-52hjeb-Item en8mxva4"
            >
              <a
                className="css-irwgy-ListLink en8mxva5"
                href="#discover-send-receive"
              >
                Send & Receive
              </a>
            </li>
            <li
              className="css-52hjeb-Item en8mxva4"
            >
              <a
                className="css-irwgy-ListLink en8mxva5"
                href="#facilitate"
              >
                Facilitate
              </a>
            </li>
            <li
              className="css-52hjeb-Item en8mxva4"
            >
              <a
                className="css-irwgy-ListLink en8mxva5"
                href="#me-to-me"
              >
                Me to Me
              </a>
            </li>
          </ul>
        </div>
        <div
          className="css-872axt-Group en8mxva1"
        >
          <h3
            className="css-p408ob-Heading en8mxva2"
          >
            Concepts
          </h3>
          <ul
            className="css-499foe-List en8mxva3"
          >
            <li
              className="css-52hjeb-Item en8mxva4"
            >
              <a
                className="css-irwgy-ListLink en8mxva5"
                href="#concepts-basics"
              >
                Basics
              </a>
            </li>
            <li
              className="css-52hjeb-Item en8mxva4"
            >
              <a
                className="css-irwgy-ListLink en8mxva5"
                href="#concepts-essentials"
              >
                Essentials
              </a>
            </li>
            <li
              className="css-52hjeb-Item en8mxva4"
            >
              <a
                className="css-irwgy-ListLink en8mxva5"
                href="#concepts-integrations"
              >
                Integrations
              </a>
            </li>
          </ul>
        </div>
        <div
          className="css-872axt-Group en8mxva1"
        >
          <h3
            className="css-p408ob-Heading en8mxva2"
          >
            Guides
          </h3>
          <ul
            className="css-499foe-List en8mxva3"
          >
            <li
              className="css-52hjeb-Item en8mxva4"
            >
              <a
                className="css-irwgy-ListLink en8mxva5"
                href="#guides-prerequisites"
              >
                Prerequisites
              </a>
            </li>
            <li
              className="css-52hjeb-Item en8mxva4"
            >
              <a
                className="css-irwgy-ListLink en8mxva5"
                href="#guides-get-building"
              >
                Get Building
              </a>
            </li>
            <li
              className="css-52hjeb-Item en8mxva4"
            >
              <a
                className="css-irwgy-ListLink en8mxva5"
                href="#features"
              >
                Features
              </a>
            </li>
          </ul>
        </div>
        <div
          className="css-872axt-Group en8mxva1"
        >
          <h3
            className="css-p408ob-Heading en8mxva2"
          >
            API Reference
          </h3>
          <ul
            className="css-499foe-List en8mxva3"
          >
            <li
              className="css-52hjeb-Item en8mxva4"
            >
              <a
                className="css-irwgy-ListLink en8mxva5"
                href="#api-reference-documentation"
              >
                Documentation
              </a>
            </li>
            <li
              className="css-52hjeb-Item en8mxva4"
            >
              <a
                className="css-irwgy-ListLink en8mxva5"
                href="api-reference-sdk-support"
              >
                SDK Support
              </a>
            </li>
            <li
              className="css-52hjeb-Item en8mxva4"
            >
              <a
                className="css-irwgy-ListLink en8mxva5"
                href="#api-reference-what-it-do"
              >
                What does the Dwolla API do?
              </a>
            </li>
          </ul>
        </div>
        <div
          className="css-872axt-Group en8mxva1"
        >
          <h3
            className="css-p408ob-Heading en8mxva2"
          >
            Tools and SDKs
          </h3>
          <ul
            className="css-499foe-List en8mxva3"
          >
            <li
              className="css-52hjeb-Item en8mxva4"
            >
              <a
                className="css-irwgy-ListLink en8mxva5"
                href="#tools-and-sdks-ruby"
              >
                Ruby
              </a>
            </li>
            <li
              className="css-52hjeb-Item en8mxva4"
            >
              <a
                className="css-irwgy-ListLink en8mxva5"
                href="#tools-and-sdks-python"
              >
                Python
              </a>
            </li>
            <li
              className="css-52hjeb-Item en8mxva4"
            >
              <a
                className="css-irwgy-ListLink en8mxva5"
                href="#tools-and-sdks-php"
              >
                PHP
              </a>
            </li>
            <li
              className="css-52hjeb-Item en8mxva4"
            >
              <a
                className="css-irwgy-ListLink en8mxva5"
                href="#tools-and-sdks-csharp"
              >
                C#
              </a>
            </li>
            <li
              className="css-52hjeb-Item en8mxva4"
            >
              <a
                className="css-irwgy-ListLink en8mxva5"
                href="#tools-and-sdks-java"
              >
                Java
              </a>
            </li>
            <li
              className="css-52hjeb-Item en8mxva4"
            >
              <a
                className="css-irwgy-ListLink en8mxva5"
                href="#tools-and-sdks-javascript"
              >
                JavaScript
              </a>
            </li>
          </ul>
        </div>
      </div>
    </div>
  </div>,
]
`;
